rosetta-source/src/main/rosetta/product-template-type.rosetta

```diff
@@ -23,15 +23,17 @@ import cdm.product.collateral.*
 
 import cdm.mapping.config.*
 
-type TransferableProduct:  <"A TransferableProduct is a type of Product which can be used in a SettlementCommitment for a basic cash settled trade of either an Asset with or without the addition of specific EconomicTerms.">
-    asset Asset (1..1)
-    economicTerms EconomicTerms (0..1)
+type ProductBase: <"Serves as an abstract class to specify a product using a productIdentifier.">
+    productTaxonomy ProductTaxonomy (0..*) <"Specifies the product taxonomy, which is composed of a taxonomy value and a taxonomy source.">
+    economicTerms EconomicTerms (1..1)  <"The price forming features, including payouts and provisions.">
```

Comment:
Now that `TransferableProduct` no longer extends `ProductBase`, consider keeping `productIdentifier` in `ProductBase`. See also comment below regarding naming.

For P3, you can probably do away with `ProductBase` altogether and replace by `NonTransferableProduct` (and `ContractualProduct` extends it with a legal agreement).

Reply:
P3